### PR TITLE
Add concurrency to test-ipsec-upgrade

### PR DESCRIPTION
**Pull request by aanm**
The changes on this PR helps to decrease the test run time significantly. During testing if this was possible, a lot of flakes were being caused by the egress-gateway, egress-gateway-with-l7-policy and from-cidr-host-netns tests which is why they run separately without concurrency and then all the other tests run parallel afterwards.

You can find 10 successful and unsuccessful runs in https://github.com/cilium/cilium/actions/runs/11496595336

**Note**: The failed runs from the link above were all caused due short timeouts which is fixed by the last commit.

| Name | Before    | After     | Time Decreased |
|-----------|-----------|-----------|----------------|
| Test 1    | 00:49:48  | 00:29:54  | -39.96%        |
| Test 2    | 00:00:23  | 00:00:19  | -17.39%        |
| Test 3    | 00:49:08  | 00:25:04  | -48.98%        |
| Test 4    | 00:00:23  | 00:00:21  | -8.7%        |
| Test 5    | 00:49:18  | 00:26:30  | -46.25%        |
| Test 6    | 00:00:22  | 00:00:23  | 4.55%        |
| Test 7    | 00:44:46  | 00:21:20  | -52.35%        |
| Test 8    | 00:00:19  | 00:00:21  | 10.35%        |
| Test 9    | 00:49:34  | 00:25:33  | -48.45%        |
| Test 10   | 00:00:22  | 00:00:19  | -13.64%        |

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

File: cilium-cli/connectivity/builder/from_cidr_host_netns.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 type fromCidrHostNetns struct{}
 
 func (t fromCidrHostNetns) build(ct *check.ConnectivityTest, templates map[string]string) {
-	newTest("from-cidr-host-netns", ct).
```
Comment (pchaigno) [inline cilium-cli/connectivity/builder/from_cidr_host_netns.go]: Do we know why these three tests specifically are failing when run in parallel with other tests?

File: cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ var clientEgressL7HTTPExternalYAML string
 type egressGatewayWithL7Policy struct{}
 
 func (t egressGatewayWithL7Policy) build(ct *check.ConnectivityTest, templates map[string]string) {
-	newTest("egress-gateway-with-l7-policy", ct).
```
Comment (pchaigno) [inline cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go]: How come this is an issue? We don't run this test in the IPsec workflows given we don't enable egress gateway there today... :confused:

**Review by pchaigno — COMMENTED** [review]
Looks good, but I'm quite confused about how egress gateway tests could be an issue in the IPsec workflow... given we don't run them there.

**Comment by pchaigno** [comment]
> > Looks good, but I'm quite confused about how egress gateway tests could be an issue in the IPsec workflow... given we don't run them there.
> 
> Ok, if we don't run them here then I can drop this commit. I wasn't sure about it as I copied this commit from a similar concurrent PR (#34806) and it didn't matter which PR would be merged first.

Ack for the egress gateway tests, but the `from-cidr-host-netns` I believe is executed in the IPsec workflows.

**Comment by pchaigno** [comment]
Hm, the failure in [the IPsec upgrade tests](https://github.com/cilium/cilium/actions/runs/11374083703/job/31642216481) feels a bit suspicious:
```
🟥 failed to flush ct entries: %w command failed (pod=kube-system/cilium-qq2x7, container=cilium-agent): "time=\"2024-10-16T21:40:39.620403221Z\" level=error msg=\"Unable to delete CT entry\" error=\"unable to delete element 10.244.2.53:61575 --> 10.244.2.84:62751, 6, 1 from map cilium_ct4_global: delete: key does not exist\" key=\"10.244.2.53:61575 --> 10.244.2.84:62751, 6, 1\" subsys=map-ct\n"
```
I haven't seen this failure before, so wondering if it could be related to the changes made here. Isn't it possible we're trying to flush the CT map in parallel and therefore failing?

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 3]
```diff
@@ -360,12 +377,26 @@ jobs:
           kubectl get pods --all-namespaces -o wide
           kubectl -n kube-system exec daemonset/cilium -c cilium-agent -- cilium-dbg status
 
-      - name: Check conn-disrupt-test after downgrading
+      - name: Run connection interrupted tests after downgrading (${{ join(matrix.*, ', ') }})
+        if: ${{ steps.vars.outputs.downgrade_version != '' }}
+        uses: ./.github/actions/conn-disrupt-test-check
+        with:
+          job-name: cilium-downgrade-${{ matrix.name }}
```
Comment (pchaigno) [inline .github/workflows/tests-ipsec-upgrade.yaml:384]: Could you explain in the commit description why this change is needed? I think the `seq-` / `!seq-` are likely self-explanatory for most, but it's harder to guess why the conn-disrupt test wouldn't be parallelizable.

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/tests-ipsec-upgrade.yaml (hunk 3 above)
Comment (aanm) [reply .github/workflows/tests-ipsec-upgrade.yaml:384]: Honestly there is no special reason. It was suggested by Martynas that we should split the steps to make it more obvious that the conn-disrupt tests are an "important" part to check after an upgrade / downgrade. This was actually not obvious when I was adding concurrency to the e2e-upgrade tests as I got confused which tests were checking for the conn-disrupt and which tests weren't. Thus, I just applied the same logic in here.

Now, if you ask why aren't we running the `conn-disrupt` in parallel? I would say it's simply because there's only, as far as I know, one test executed and therefore it doesn't make sense to run in parallel.

Comment (pchaigno) [reply .github/workflows/tests-ipsec-upgrade.yaml:384]: Ah. So we still don't know why https://github.com/cilium/cilium/pull/35362#issuecomment-2419354256 happened?

Comment (aanm) [reply .github/workflows/tests-ipsec-upgrade.yaml:384]: I haven't seen it happening since I rebased. But your explanation in the comment would make more sense indeed.
